uploadgui/src/main/scala/se/lu/nateko/cp/meta/upload/Form.scala

```diff
@@ -15,46 +15,40 @@ class Form(
 	onSubmitterSelect: SubmitterProfile => Future[IndexedSeq[Station]]
 )(implicit envri: Envri.Envri, envriConf: EnvriConfig) {
 
-	var formType: Option[String] = None
 	val dataElements = new HtmlElements(".data-section")
 	val collectionElements = new HtmlElements(".collection-section")
-	val onFileTypeSelected: String => Unit = (fileType: String) => {
-		fileType match {
-			case "data" => {
-				dataElements.show()
-				collectionElements.hide()
-				fileInput.enable()
-			}
-			case "document" => {
-				dataElements.hide()
-				collectionElements.hide()
-				fileInput.enable()
-			}
-			case "collection" => {
-				dataElements.hide()
+
+	val onFileTypeSelected: String => Unit = (_) => {
+		dataElements.hide()
+		collectionElements.hide()
+		fileInput.enable()
+
+		typeControl.formType match {
+			case Data => dataElements.show()
+			case Collection => {
 				collectionElements.show()
 				fileInput.disable()
 			}
+			case _ =>
 		}
 		updateButton()
 	}
 
-	def submitAction(): Unit = typeControl.value match {
-		case Some("data") =>
+	def submitAction(): Unit = typeControl.formType match {
+		case Data =>
 			for(dto <- dto; file <- fileInput.file; nRows <- nRowsInput.value; spec <- objSpecSelect.value) {
 				whenDone(Backend.tryIngestion(file, spec, nRows)){ _ =>
 					onUpload(dto, Some(file))
 				}
 			}
-		case Some("document") =>
-			for(dto <- dto; file <- fileInput.file) {
-				onUpload(dto, Some(file))
-			}
-		case Some("collection") =>
+		case Collection =>
 			for(dto <- dto) {
 				onUpload(dto, None)
 			}
-		case _ => ()
+		case _ =>
```

Comment:
Better not to use catch-all here and in other similar places, but to instead rely on compiler's pattern-match-exhaustion check.